songsuro/condition/model.py

```diff
@@ -24,24 +23,32 @@ def __init__(
 		self.enhanced_condition_encoder = FFTEncoder()
 		self.prior_estimator = PriorEstimator(hidden_size, prior_output_dim)
 
-	def forward(self, lyrics, quantized_f0):
+	def forward(
+		self,
+		lyrics,
+		lyrics_lengths: torch.Tensor,
+		quantized_f0,
+		quantized_f0_lengths: torch.Tensor,
+	):
 		"""
 		Forward pass of the conditional encoder.
 		Given original audio and the lyrics, it generates embedding vectors with timbre, lyrics, melody and style.
 
 		:param lyrics: Tokenized lyrics sequence. Should be a tensor or can be converted to one.
+		:param lyrics_lengths: Lengths of the lyrics sequences. [batch_size x Seq_len]
 		:param quantized_f0: Pre-processed quantized f0 data.
+		:param quantized_f0_lengths: Lengths of the quantized f0 sequences. [batch_size x max_length]
```

Comment:
I think the lengths tensor shape will be just [batch size] ??
For example,
torch.Tensor([32,35,40,29])